### PR TITLE
Fix fcitx5

### LINKED ISSUE #109631: fctix5 does not work in qt programs
**Describe the bug**

fcitx5 does not emerge after pressed <kbd>Ctrl</kbd> + <kbd>Space</kbd> in qt applications.

Relevant information from `fcitx5-diagnose`:
```
**Cannot find fcitx5 input method module for Qt5.**
``` 

**Additional context**

IIUC, fcitx5-qt should provide usr/lib/qt/plugins/platforminputcontexts/libfcitx5platforminputcontextplugin.so as a qt plugin. However, the current output does not contain this file:

```
/nix/store/xaxdimay7izbr66kkfm26qwyw0zwb08q-fcitx5-qt-5.0.1
├── include
│   └── Fcitx5Qt5
│       ├── Fcitx5Qt5DBusAddons
│       │   ├── fcitx5qt5dbusaddons_export.h
│       │   ├── fcitxqtcontrollerproxy.h
│       │   ├── fcitxqtdbustypes.h
│       │   ├── fcitxqtinputcontextproxy.h
│       │   ├── fcitxqtinputmethodproxy.h
│       │   └── fcitxqtwatcher.h
│       ├── fcitx5qt5dbusaddons_version.h
│       ├── fcitx5qt5widgetsaddons_version.h
│       └── Fcitx5QtWidgetsAddons
│           ├── fcitx5qt5widgetsaddons_export.h
│           ├── fcitxqtconfiguifactory.h
│           ├── fcitxqtconfiguiplugin.h
│           ├── fcitxqtconfiguiwidget.h
│           ├── fcitxqti18nhelper.h
│           └── fcitxqtkeysequencewidget.h
├── lib
│   ├── cmake
│   │   ├── Fcitx5Qt5DBusAddons
│   │   │   ├── Fcitx5Qt5DBusAddonsConfig.cmake
│   │   │   ├── Fcitx5Qt5DBusAddonsConfigVersion.cmake
│   │   │   ├── Fcitx5Qt5DBusAddonsTargets.cmake
│   │   │   └── Fcitx5Qt5DBusAddonsTargets-release.cmake
│   │   └── Fcitx5Qt5WidgetsAddons
│   │       ├── Fcitx5Qt5WidgetsAddonsConfig.cmake
│   │       ├── Fcitx5Qt5WidgetsAddonsConfigVersion.cmake
│   │       ├── Fcitx5Qt5WidgetsAddonsTargets.cmake
│   │       └── Fcitx5Qt5WidgetsAddonsTargets-release.cmake
│   ├── fcitx5
│   │   ├── libexec
│   │   │   └── fcitx5-qt5-gui-wrapper
│   │   └── qt5
│   │       └── libfcitx-quickphrase-editor5.so
│   ├── libFcitx5Qt5DBusAddons.so -> libFcitx5Qt5DBusAddons.so.1
│   ├── libFcitx5Qt5DBusAddons.so.1 -> libFcitx5Qt5DBusAddons.so.5.0.1
│   ├── libFcitx5Qt5DBusAddons.so.5.0.1
│   ├── libFcitx5Qt5WidgetsAddons.so -> libFcitx5Qt5WidgetsAddons.so.2
│   ├── libFcitx5Qt5WidgetsAddons.so.2 -> libFcitx5Qt5WidgetsAddons.so.5.0.1
│   └── libFcitx5Qt5WidgetsAddons.so.5.0.1
└── share
    └── locale
        ├── ca
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        ├── da
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        ├── de
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        ├── es
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        ├── fr
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        ├── he
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        ├── ja
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        ├── ko
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        ├── ru
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        ├── zh_CN
        │   └── LC_MESSAGES
        │       └── fcitx5-qt.mo
        └── zh_TW
            └── LC_MESSAGES
                └── fcitx5-qt.mo

35 directories, 41 files
```

**Notify maintainers**

@poscat0x04 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.89, NixOS, 21.03.20210115.b3616bd (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20201205_a5d85d0`
 - channels(berberman): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - libsForQt5.fcitx5-qt
  - libsForQt512.fcitx5-qt
  - libsForQt514.fcitx5-qt
  - fcitx5-chinese-addons
  - fcitx5-with-addons
  - fcitx5-configtool
# a list of nixos modules affected by the problem
module:
  - i18n.inputMethod
  - i18n.inputMethod.fcitx5
```

### PR DESCRIPTION
**Pull request by berberman**
###### Motivation for this change
Closes #109631

This PR fixes two bugs in fcitx5 packaging:

* libfcitx5platforminputcontextplugin.so was not captured, resulting in missing input method module for Qt5
* .desktop files pointed to unwrapped executables before symlinkJoin, leading to failure of addons  

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @poscat0x04

### CODE REVIEW

File: pkgs/tools/inputmethods/fcitx5/with-addons.nix [hunk 1]
```diff
@@ -12,6 +12,14 @@ symlinkJoin {
...
       --suffix XDG_DATA_DIRS : "$out/share" \
       --suffix PATH : "$out/bin"
+
+    desktop=share/applications/org.fcitx.Fcitx5.desktop
+    autostart=etc/xdg/autostart/org.fcitx.Fcitx5.desktop
+    rm $out/$desktop
+    rm $out/$autostart
+    cp ${fcitx5}/$desktop $out/$desktop
+    sed -i $out/$desktop -e "s|^Exec=.*|Exec=$out/bin/fcitx5|g"
+    cp $out/$desktop $out/$autostart
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/fcitx5/with-addons.nix:22]:
```suggestion
    ln -s $out/$desktop $out/$autostart
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109649` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>fcitx5-chinese-addons: log <a href="https://termbin.com/d7g6">https://termbin.com/d7g6</a></li>
  </ul>
</details>
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>fcitx5-configtool</li>
<li>fcitx5-with-addons</li>
<li>libsForQt5.fcitx5-qt (libsForQt515.fcitx5-qt)</li>
<li>libsForQt512.fcitx5-qt</li>
<li>libsForQt514.fcitx5-qt</li>
</ul>
</details>

**Review by poscat0x04 — APPROVED** [review]
LGTM

**Comment by berberman** [comment]
> 1 package failed to build and are new build failure:
> * fcitx5-chinese-addons: log https://termbin.com/d7g6

@SuperSandro2000  I could not reproduce this error on my machine.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109649` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>fcitx5-chinese-addons</li>
<li>fcitx5-configtool</li>
<li>fcitx5-with-addons</li>
<li>libsForQt5.fcitx5-qt (libsForQt515.fcitx5-qt)</li>
<li>libsForQt512.fcitx5-qt</li>
<li>libsForQt514.fcitx5-qt</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
Please remove the master merge commit. Then we can merge this.

**Comment by berberman** [comment]
ping @SuperSandro2000

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109649` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>fcitx5-chinese-addons</li>
<li>fcitx5-configtool</li>
<li>fcitx5-with-addons</li>
<li>libsForQt5.fcitx5-qt (libsForQt515.fcitx5-qt ,plasma5Packages.fcitx5-qt)</li>
<li>libsForQt512.fcitx5-qt</li>
<li>libsForQt514.fcitx5-qt</li>
</ul>
</details>

**Pull request merged**